src/anchorcommand.js

```diff
@@ -207,6 +207,12 @@ export default class AnchorCommand extends Command {
 				[ 'anchorId', ...truthyManualDecorators, ...falsyManualDecorators ].forEach( item => {
 					writer.removeSelectionAttribute( item );
 				} );
+			} else if (selection.getSelectedElement()?.name === 'anchor') {
+				// Replace an invisible anchor.
```

Comment:
Need to add the logic to remove the anchor if user has saved the empty ID while updating
```
// Replace an invisible anchor.
        if(id) {
          const anchor = writer.createElement('anchor', {
            anchorId: id
          });
          model.insertObject(anchor, null, null, {setSelection: 'on'});
        }
        else {
          // Remove an invisible anchor.
          if (selection.getSelectedElement()?.name === 'anchor') {
            model.deleteContent(selection);
          }
        }
```

Reply:
Actually this is the same behaviour as visible anchors. I think this should be moved to a separate issue, and I think we can simply add the `required` attribute on the input element.